DatamartManagementService/DatamartManagementService.Infrastructure/Persistence/PayrollRepository.cs

```diff
@@ -0,0 +1,133 @@
+﻿using DatamartManagementService.Infrastructure.Persistence.Entities;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DatamartManagementService.Infrastructure.Persistence
+{
+    public interface IPayrollRepository
+    {
+        Task AddEmployeePayroll(EmployeePayroll newPayroll);

```

Comment:
What did we learn in our refactoring?
Or the I in SOLID.

Reply:
Also, employee payroll and employee payroll details can be seen as two separate things.

Reply:
Of course, you can always refactor after. But since you already know about I, why not just do it first?